examples/chip-tool/commands/casting/DiscoverCommissionersCommand.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "DiscoverCommissionersCommand.h"
+#include <arpa/inet.h>
+#include <chrono>
+#include <string>
+#include <thread>
+
+using namespace ::chip;
+
+CHIP_ERROR DiscoverCommissionersCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
+{
+    mCommissionableNode.Init();
+    mCommissionableNode.DiscoverAllCommissioners();
+
+    uint32_t waitTime     = 500;
+    System::Timer * timer = nullptr;
+    if (DeviceLayer::SystemLayer.NewTimer(timer) == CHIP_NO_ERROR)
+    {
+        timer->Start(
+            waitTime, [](System::Layer *, void *, System::Error err) { DeviceLayer::PlatformMgr().Shutdown(); }, nullptr);
+    }
+    else
+    {
+        ChipLogError(chipTool, "Failed to create the shutdown timer. Kill with ^C.\n");
```

Comment:
I believe the log does not require a final \n. Do you explicitly want extra separation?

Reply:
Removed the extra \n